systemservices/ethwallet/ethwallet/sign.go

```diff
@@ -0,0 +1,63 @@
+package ethwallet
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/mesg-foundation/core/client/service"
+	"github.com/mesg-foundation/core/systemservices/ethwallet/x/xgo-ethereum/xaccounts"
+)
+
+type signInputs struct {
+	Address     string       `json:"address"`
+	Passphrase  string       `json:"passphrase"`
+	Transaction *transaction `json:"transaction"`
+	ChainID     int64        `json:"chainID"`
+}
+
+type transaction struct {
```

Comment:
https://github.com/ethereum/go-ethereum/blob/c1c43011211e70ffdd2a78fb1a467adb5db9f65e/core/types/transaction.go#L46

- Replace `Value` and `GasPrice with big.Int (it implemnts unmarshal json interface https://golang.org/pkg/math/big/#Int.UnmarshalJSON)
- rename `Data` json tag to `"input"`
- rename `Data` to either `Payload` or `Input` 
- rename `Value` To `Amount` (not mandatory)
- rename `To` -> `Recipient` (not mandatory)

Reply:
Because of web3 transaction definition: https://web3js.readthedocs.io/en/1.0/web3-eth.html#eth-sendtransaction

Reply:
Ok so this is sign transaction, my bad.

https://github.com/ethereum/go-ethereum/blob/d3441ebb563439bac0837d70591f92e2c6080303/signer/core/types.go#L95:6

So there `GasLimit` Should be changed to `Gas` then, because it is gas in web3 and go-ethereum.

Also the field From is missing but I think it's irrelevant.